pkg/applicationserver/io/io.go

```diff
@@ -82,19 +88,22 @@ var errBufferFull = errors.DefineResourceExhausted("buffer_full", "buffer is ful
 
 // SendUp sends an upstream message.
 // This method returns immediately, returning nil if the message is buffered, or with an error when the buffer is full.
-func (s *Subscription) SendUp(up *ttnpb.ApplicationUp) error {
+func (s *Subscription) SendUp(ctx context.Context, up *ttnpb.ApplicationUp) error {
 	select {
 	case <-s.ctx.Done():
 		return s.ctx.Err()
-	case s.upCh <- up:
+	case s.upCh <- &ContextualApplicationUp{
+		Context:       ctx,
```

Comment:
It is important to note that this context is likely to be canceled shortly after `SendUp` returns, so if this (or a derived) context is used any future requests (gRPC, HTTP), those may return a `context.Canceled` error without even making the request.

cc: @johanstokking